### DISCUSSION TITLE
Extracting data

### DISCUSSION THREAD

**Discussion by davidmoranccm** [ARM-DOE/pyart#1316] [General]
I’m a meteorologist who’s been programming in Python for a little over a year, but only recently started looking at PyART. I have a problem I’m not sure if I can use with PyART and wanted to ask.

I have two weeks’ worth of radar data from a NEXRAD site. I need to know the total rainfall at a certain lat/lon over the two week period. Can this be done with PyART?

Thanks!
David

**Comment by zssherman** [comment]
Hi @davidmoranccm, you could always try to do a a loop with pyarts nexrad reader, and use one of the estimate rain rate functions in pyarts retrieval module. For extracting at a specific lat and lon, I think the column sect might work after the rain rate retrieval, if not, once things get ported to xarray the .sel will help.

**Comment by davidmoranccm** [comment]
Thanks, Zach! I’ll play around with it later.

David

David Moran, CCM
AMS Certified Consulting Meteorologist
E-Mail: ***@***.***
Phone: (405) 693-5632

**Comment by wxwolff** [comment]
Just a thought if you are familiar with Xarray…

For each volume scan, use the radar.to_grid method and then grid.to_xarray.
Create an empty lists, gr_list, and append each grid.
Use xarray.concat(gr_list) to create an Xarray dataset

From there, you can use Xarray’s powerful tools to calculate aggregate statistics.

D


_____________________________________
David B. Wolff, Code 612
GPM Deputy Project Scientist for GV
GPM Ground Validation System Manager
NASA Wallops Flight Facility
34200 Fulton Street
Building N159, Room E220
Wallops Island, VA 23337
Ph: 757-824-1515; Cell: 443-262-1146


From: davidmoranccm ***@***.***>
Reply-To: ARM-DOE/pyart ***@***.***>
Date: Friday, November 4, 2022 at 7:22 PM
To: ARM-DOE/pyart ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] Re: [ARM-DOE/pyart] Extracting data (Discussion #1316)

Thanks, Zach! I’ll play around with it later.

David

David Moran, CCM
AMS Certified Consulting Meteorologist
E-Mail: ***@***.***
Phone: (405) 693-5632

**Comment by scollis** [comment]
Also for general QPE questions highly suggest joining https://openradar.discourse.group/
